cmake/HPX_SetupBoost.cmake

```diff
@@ -138,7 +140,7 @@ endif()
 
 include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
 link_directories(${Boost_LIBRARY_DIRS})
-if(NOT MSVC)
+if((NOT MSVC) OR HPX_WITH_VCPKG)
```

Comment:
This is not only related whether the Boost libraries are included explicitly or not. This also changes the command line options passed to the compiler (see explanations above). I don't see a way to do this without introducing a special configuration variable.